### DISCUSSION TITLE
Migration help (earlier versions)

### DISCUSSION THREAD

**Discussion by mpf82** [vitmalina/w2ui#2015] [General]
# 1.4.x -> 1.5rc1:

see issue https://github.com/vitmalina/w2ui/issues/1351

---

# 1.5rc1 -> 1.5 final

**The following properties are deprecated and you will see a notice in the browser's dev console if you are still using them:**

w2form:

    tab.caption         -> tab.text
    tab.hint            -> tab.tooltip
    field.name          -> field.field
    field.html.caption  -> field.html.label
    action.caption      -> action.text

w2grid:

    search.caption      -> search.label
    columnGroup.caption -> columnGroup.text
    column.caption      -> column.text

w2layout:

    layout.content      -> layout.html
    panel.content       -> panel.html

w2toolbar:

    item.caption        -> item.text
    item.hint           -> item.tooltip

w2sidebar:

    node.caption        -> node.text

### Other Changes and Features:

- w2field
    - w2field 'color' has a new color picker and supports alpha channel
    (but alpha channel preview is broken if jquery-ui is loaded)\
    http://w2ui.com/web/demos/#!fields/fields-8
    - w2field 'color' no longer has the '#' prefix as a default - if we want it, we must set it manually:\
    ``field.options : { prefix : '#' },``
    - w2field 'checkbox' now has the caption to the right of the checkbox
    https://github.com/vitmalina/w2ui/issues/1947 (CLOSED)
    to restore old behavior ``field.html.text`` must be used instead of ``field.html.label``
    - w2field.equalto check has been dropped
- w2form
    - form fields can have a column span:\
    http://w2ui.com/web/demos/#!forms/forms/11
    - new methods: ``enable, disable, show, hide`` to manipulate multiple fields at once
    - new field types: ``html, empty, custom, check, checks (alias), map, array``
    - added event ``onProgress``
    - added event ``onInput``
    - new property ``httpHeaders`` to provide headers in AJAX calls
    - new property  ``method`` to override w2utils.settings.dataType
    - added field.html.anchor
    - added added field.html.groupStyle, field.html.groupTitleStyle, field.html.groupCollapsible
- w2grid
    - ``w2grid.column.clipboardCopy``
        adds a copy icon next to the value in the column
    - in the grid's delete-record-confirm-dialog, the "Delete" Button no longer has the class "w2ui-btn-red"
    - w2grid no longer sends "get" as a default command, and "delete" / "save" have been moved from "cmd" to "action".\
    https://github.com/vitmalina/w2ui/blob/master/src/w2grid.js#L2491
    https://github.com/vitmalina/w2ui/issues/1869
    https://github.com/vitmalina/w2ui/issues/1837#issuecomment-504672005
- w2toolbar
    - new item.type: 'new-line'
- translations
    - some translations have changed:
        ```
        "Are you sure you want to delete selected records?" -> 'Are you sure you want to delete NN records?'
        ```
    - some translations have been added:
        ```
        "Copied"
        "Copy to clipboard"
        "Maximum total size is"
        "Maximum number of files is"
        "Field should be at least NN characters."
        ```

---

# 1.5 final -> 2.0 (WIP)

As 2.0 is currently in development, this section will be updated as development progresses.

**Note: most of the properties that have been marked as deprecated in 1.5 are now removed!**

w2layout:

- layout.content - deprecated
- panel.callback        -> panel.removed
- panel.onContent       -> panel.onChange

w2field:

- w2field.normMenu      -> w2utils.normMenu
- ``$().w2field('addType', 'myType', function (options) {`` is now ``addType( 'myType', function(options) {``
- enum fields: defining ``onNew`` is no longer enough to add arbitrary items. You need to set ``options.autoAdd: true`` now.

w2grid:

- ``recordHeight``: default changed from 24 to 32 ( #914e2be3 )

w2form:

- ``focus`` can be a string
- ``applyFocus()`` takes an optional string/integer argument and returnd the focused element (or ``undefined`` if the index/field could not be focused)

w2popup:

- added ``openMaximized``

w2utils:

- Accessing "missing" translations via ``w2utils.lang`` will by default result in a warning being logged to the console. You can change this behavior in the w2uils settings.

Translations added:

    'Advanced Search'
    'Do you want to delete "${item}" search item?'
    'Load ${count} more...'
    'Maximum number of files is ${count}'
    'Maximum total size is ${count}'
    'record'
    'records'
    'Request aborted.'
    'Saved Searches'
    'Search took ${count} seconds'
    'Sorting took ${count} seconds'

Translations removed:

    'Clear Search'
    'Delete Confirmation'
    'Edit Record'
    'Field should be equal to '
    'Load'
    'More'
    'Open Search Fields'
    'Records'
    'Reset Column Size'
    'Search took'
    'Search...'
    'sec'
    'Select Search Field'
    'Sorting took'
    'Toggle Line Numbers'

Translations changed:

    'Are you sure you want to delete NN records?' -> 'Are you sure you want to delete ${count} ${records}?'
    'Field should be at least NN characters.' -> 'Field should be at least ${count} characters.'
    'letters or more...' -> '${count} letters or more...'
    'Server Response' + 'sec' -> 'Server Response ${count} seconds'

**Comment by vitmalina** [comment]
More current 1.5 -> 2.0 upgrade guide is here https://github.com/vitmalina/w2ui/discussions/2403
